lib/_http_outgoing.js

```diff
@@ -311,6 +318,14 @@ function _storeHeader(firstLine, headers) {
   if (state.sentExpect) this._send('');
 }
 
+var fastTest = function (field, expression, expressionLength) {
```

Comment:
Can you write this as `function fastTest(...) {`?  Also, can you post some benchmark numbers that demonstrate it's actually faster?  I have a hard time believing V8's regex engine doesn't already perform this optimization.